api/v1alpha2/contentlibraryitem_types.go

```diff
@@ -0,0 +1,309 @@
+	CertChain []string `json:"certChain,omitempty"`
+}
+
+// FileInfo represents the information of a file in a content library item in
+// vCenter.
+type FileInfo struct {
+	// Name describes the name of the file.
+	Name string `json:"name"`
+
+	// +optional
+
+	// SizeInBytes describes the total number of bytes used by the file on disk.
+	SizeInBytes resource.Quantity `json:"sizeInBytes,omitempty"`
+
+	// +optional
+
+	// Version describes the version of the library item file.
+	// This value is incremented when the file is modified on disk.
+	Version string `json:"version,omitempty"`
+
+	// +optional
+
+	// Cached describes whether or not the file is available locally on disk.
+	Cached bool `json:"cached,omitempty"`
+
+	// +optional
+
+	// StorageURI describes the fully-qualified path to the file on a datastore,
+	// ex. "[my-datastore-1] library-1/item-1/file1.vmdk".
+	// This URL is useful for creating a device that is backed by this file
+	// (i.e. mounting an ISO file via a virtual CD-ROM device).
+	StorageURI string `json:"storageURI,omitempty"`
+}
+
+// ContentLibraryItemSpec defines the desired state of a ContentLibraryItem.
+type ContentLibraryItemSpec struct {
+	// ID describes the unique identifier used to find the library item in
+	// vCenter.
+	//
+	// Please note this value depends on the type of the library to which this
+	// item belongs:
+	// - LibraryType=ContentLibrary -- ID is a content library item UUID.
+	// - LibraryType=Inventory      -- ID is a vSphere VM managed object ID.
+	ID string `json:"id"`
+
+	// +required
+
+	// LibraryName describes the name of the library to which this item belongs.
+	// Namespace-scoped items belong to namespace-scoped libraries and
+	// cluster-scoped items belong to cluster-scoped libraries.
+	LibraryName string `json:"libraryName"`
+}
+
+// ContentLibraryItemStatus defines the observed state of ContentLibraryItem.
+type ContentLibraryItemStatus struct {
+	// +optional
+
+	// Name describes the name of the underlying item in vCenter.
+	Name string `json:"name,omitempty"`
+
+	// +optional
+
+	// Description is a human-readable description for this library item.
+	Description string `json:"description,omitempty"`
+
+	// +optional
+
+	// Version describes a value that tracks changes to the library item's
+	// metadata, such as its name or description.
+	//
+	// This field is only valid for items with spec.libraryType=ContentLibrary.
+	Version string `json:"version,omitempty"`
+
+	// +optional
+
+	// ContentVersion describes a value that tracks changes to the library
+	// item's content.
+	//
+	// This field is only valid for items with spec.libraryType=ContentLibrary.
+	ContentVersion string `json:"contentVersion,omitempty"`
+
+	// +optional
+
+	// Type describes the type of the library item.
+	Type ContentLibraryItemType `json:"type,omitempty"`
+
+	// +optional
+
+	// SourceID describes a unique piece of information used to identify a
+	// library item on the remote side a subscribed library.
```

Comment:
Any plan to use this one for items from inventory libraries too? if not, may want to comment it only for the library type ContentLibrary